tikv/server.go

```diff
@@ -515,7 +534,16 @@ func (svr *Server) Coprocessor(ctx context.Context, req *coprocessor.Request) (*
 	if reqCtx.regErr != nil {
 		return &coprocessor.Response{RegionError: reqCtx.regErr}, nil
 	}
-	return cophandler.HandleCopRequest(reqCtx.getDBReader(), svr.mvccStore.lockStore, req), nil
+	var mppTaskHandler *cophandler.MPPTaskHandler = nil
+	if mockRegionRM, ok := svr.regionManager.(*MockRegionManager); ok {
+		mppTaskHandlerMap := mockRegionRM.getMPPTaskSet(reqCtx.storeId)
+		if th, ok := mppTaskHandlerMap[int64(req.Context.TaskId)]; ok {
+			mppTaskHandler = th
+		}
+	}
+	return cophandler.HandleCopRequestWithMPPCtx(reqCtx.getDBReader(), svr.mvccStore.lockStore, req, &cophandler.MPPCtx{
```

Comment:
Has `HandleCopRequest` been removed?
Can we call the old function if there is no mpp task?

Reply:
Not, but `HandleCopRequest` just call `HandleCopRequestWithMPPCtx` by passing `nil`  as the `MPPCtx`. Here, we do not call `HandleCopRequest`  because in the implementation, `mppHandler.HandleMPPDispatch` will call `BatchCoprocessor` or `Coprocessor` to handle the dag request, so inside `BatchCoprocessor`/`Coprocessor` there is no way to know if it is just a `BatchCoprocessor`/`Coprocessor` or actually a mpp task.